### PR TITLE
Drop logging the stack on HTTP errors

### PR DESCRIPTION
**Pull request by adreed-msft**
As discussed with Ze and John, this is an expensive operation that fills the log file far too much. PR coming to azure-pipeline-go soon with more log reductions.

### CODE REVIEW

File: ste/xferLogPolicy.go [hunk 1]
```diff
@@ -98,9 +98,6 @@ func NewRequestLogPolicyFactory(o RequestLogOptions) pipeline.Factory {
 				}
 
 				pipeline.WriteRequestWithResponse(b, prepareRequestForLogging(request), response.Response(), err)
-				if logLevel <= pipeline.LogError {
```
Comment (zezha-msft) [inline ste/xferLogPolicy.go]: We only want to remove the stack trace for HTTP errors. This is removing it altogether.

Comment (adreed-msft) [reply ste/xferLogPolicy.go]: Pardon me, misunderstood the code, thought it was exclusively for HTTP related errors.

File: ste/xferLogPolicy.go [hunk 2]
```diff
@@ -98,8 +98,8 @@ func NewRequestLogPolicyFactory(o RequestLogOptions) pipeline.Factory {
...
-					b.Write(stack()) // For errors (or lower levels), we append the stack trace (an expensive operation)
+				if logLevel <= pipeline.LogError && !httpError {
```
Comment (JohnRusk) [inline ste/xferLogPolicy.go]: Can you whack a brief comment in here, to explain why we are not logging call stacks for HTTP errors. Namely they don't add much value (if any) but they do add a lot of bulk to the log in cases such as lots of 503 responses.

Comment (adreed-msft) [reply ste/xferLogPolicy.go]: Sure thing!

**Comment by zezha-msft** [comment]
@adreed-msft please use "squash and merge", so that the result commit shows up with your name.

**Pull request merged**